src/components/IconList/IconList.js

```diff
@@ -0,0 +1,174 @@
+import React, { PureComponent, createRef } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import List from '@geops/react-ui/components/List';
+import Button from '@geops/react-ui/components/Button';
+import { FaCaretUp, FaCaretDown } from 'react-icons/fa';
+import { withTranslation } from 'react-i18next';
+import { compose } from 'lodash/fp';
+
+import './IconList.scss';
+
+const propTypes = {
+  disabled: PropTypes.bool,
+  options: PropTypes.array.isRequired,
+  selected: PropTypes.string,
+  icons: PropTypes.object.isRequired,
+
+  /**
+   * Icon size in pixels
+   */
+  iconSize: PropTypes.shape({
+    height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  }),
+
+  onSelect: PropTypes.func.isRequired,
+  displayOnTop: PropTypes.bool,
+  t: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  disabled: false,
+  selected: null,
+  iconSize: {
+    height: 16,
+    width: 42,
+  },
+  displayOnTop: false,
+};
+
+class IconList extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.ref = createRef();
+
+    this.state = {
+      selectedOption: props.selected,
+      iconListVis: false,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { selected } = this.props;
+
+    if (selected !== prevProps.selected) {
+      this.select(selected);
+    }
+  }
+
+  select(option) {
+    this.setState({
+      selectedOption: option,
+    });
+  }
+
+  renderOption(option) {
+    const { icons, t, options, iconSize } = this.props;
+
+    const opt = option || (options.length ? options[0] : null);
+
+    const label = <div className="wkp-icon-list-label">{t(opt)}</div>;
+
+    return opt && icons[opt] ? (
+      <>
+        <div className="wkp-icon-list-icon">
+          <img
+            src={icons[option]}
+            height={iconSize.height}
+            width={iconSize.width}
+            draggable="false"
+            alt={t('Kein Bildtext')}
+          />
+        </div>
+        {label}
+      </>
+    ) : (
+      label
+    );
+  }
+
+  renderList() {
+    const { displayOnTop, options, onSelect } = this.props;
+
+    const list = (
+      <List
+        items={options}
+        className="tm-list wkp-icon-list-list"
+        renderItem={option => this.renderOption(option)}
+        getItemKey={option => option}
+        onSelect={(e, option) => {
+          onSelect(option);
+          this.setState({
+            iconListVis: false,
+            selectedOption: option,
+          });
+        }}
+      />
+    );
+
+    if (displayOnTop) {
```

Comment:
@oterral I thought about adding tests. But, they would only make sense if you could make a snapshot test after clicking on the dropdown button. But I haven't found a solution, yet. @oterral any idea?